bcda/api/requests.go

```diff
@@ -242,10 +242,13 @@ func (h *Handler) JobStatus(w http.ResponseWriter, r *http.Request) {
 	job, jobKeys, err := h.Svc.GetJobAndKeys(r.Context(), uint(jobID))
 	if err != nil {
 		logger.Error(err)
-		// NOTE: This is a catch all and may not necessarily mean that the job was not found.
-		// So returning a StatusNotFound may be a misnomer
-		//In contrast to above, if the input COULD be valid, we return a not found header (404)
-		h.RespWriter.Exception(r.Context(), w, http.StatusNotFound, responseutils.DbErr, "")
+		if err == sql.ErrNoRows {
```

Comment:
Also! totally non-blocking nit, but should this use `errors.Is(err, sql.ErrNoRows)`? I see this check in the bcdaworker folder, although I assume it's just another way of writing this.

Reply:
Good catch, I'll update that + apply above; errors.Is will be more flexible since it can unwrap them.